packages/attack-http-requests/src/index.ts

```diff
@@ -0,0 +1,47 @@
+import Mitm from 'mitm';
+
+export interface HTTPAttackOptions {
+  ignoreUrlPatterns?: string[];
+}
+
+export default class HTTPAttack {
+  static configure(opts: HTTPAttackOptions): () => HTTPAttack {
+    return () => {
+      const attack = new HTTPAttack(opts);
+      return attack;
+    };
+  }
+
+  ignoreUrlPatterns: string[];
+  stopped = false;
+
+  constructor({ ignoreUrlPatterns = [] }: HTTPAttackOptions = {}) {
+    this.ignoreUrlPatterns = ignoreUrlPatterns;
+  }
+
+  start(): void {
+    const mitm = Mitm();
+
+    mitm.on('connect', (socket, opts) => {
+      // https://github.com/moll/node-mitm/issues/16
+      // eslint-disable-next-line
+
+      const requestedURL: string = opts?.host || '';
+
+      const shouldIgnore = this.ignoreUrlPatterns.some((needle: string) =>
+        requestedURL.includes(needle)
+      );
+
+      if (this.stopped || shouldIgnore) socket.bypass();
+    });
+
+    mitm.on('request', (_, res) => {
+      res.statusCode = 500;
+      res.end('Request failed');
+    });
+  }
+
+  stop(): void {
```

Comment:
Just a thought came up - can we stop listening to `mitm.on('connect')` when we `stop`? This would improve efficiency for non-chaos requests